File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/CourseVersionOutdatedError.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;

public class CourseVersionOutdatedError extends Notification {
  /**
   * Notification to be shown when the plugin is outdated with respect to the current course.
   * This represents a mismatch in the major version and is an error.
   */
  public CourseVersionOutdatedError() {
    super(PluginSettings.A_PLUS, getText("notification.CourseVersionError.title"),
        getText("notification.CourseVersionOutdatedError.content"), NotificationType.ERROR);
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/CourseVersionOutdatedError.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

class CourseVersionOutdatedError
/**
 * Notification to be shown when the plugin is outdated with respect to the current course.
 * This represents a mismatch in the major version and is an error.
 */
    : Notification(
    PluginSettings.A_PLUS, MyBundle.message("notification.CourseVersionError.title"),
    MyBundle.message("notification.CourseVersionOutdatedError.content"), NotificationType.ERROR
)
